tests/integration/test_charm.py

```diff
@@ -282,3 +282,10 @@ async def test_all_units_have_internal_users_synced(ops_test: OpsTest) -> None:
     for unit in ops_test.model.applications[APP_NAME].units:
         unit_conf = get_conf_as_dict(ops_test, unit.name, filename)
         assert leader_conf == unit_conf
+
+
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+async def test_remove_application(ops_test: OpsTest) -> None:
+    """Removes the application with a single unit."""
```

Comment:
(this test deploys 2 units not just 1) - why do we expect the removal of the application to require testing? is there an issue in this regard?

Reply:
In short, yes, we had issues with the lock and 1x unit removal.

The 2-node removal is also something good to test, as it may fall on: https://github.com/canonical/opensearch-operator/issues/243

Reply:
Okay we need to update the docstrings of the test then, as this test runs with 2 opensearch units

Reply:
@Mehdi-Bendriss yes, also added a test to deploy and remove app for a single unit as well.